src/Service/ProducerEventProcessor.cpp

```diff
@@ -318,6 +320,12 @@ void ProducerEventProcessorImpl::ProcessSystemMemoryUsage(SystemMemoryUsage* sys
   capture_event_buffer_->AddEvent(std::move(event));
 }
 
+void ProducerEventProcessorImpl::ProcessApiEvent(ApiEvent* api_event) {
+  ClientCaptureEvent event;
+  *event.mutable_api_event() = std::move(*api_event);
```

Comment:
we probably want to convert it to variable-encoding friendly proto for the service<->client communication